plugins/csv_dataset.cc

```diff
@@ -575,7 +575,13 @@ struct CsvDataset::Itl: public TabularDataStore {
         filename = config.dataFileUrl.toString();
         
         // Ask for a memory mappable stream if possible
-        ML::filter_istream stream(filename, { { "mapped", "true" } });
+        ML::filter_istream stream;
+        try {
+            stream.open(filename, { { "mapped", "true" } });
+        }
+        catch (const std::exception & e) {
+             throw HttpReturnException(400, "dataFileUrl pointed to an empty file");
```

Comment:
Two problems here:
1.  It's not at all true that this exception means the file is empty.  It could also be because it doesn't exist.
2.  You're eating the helpful error message (eg, "file not found") and replacing with another one ("dataFileUrl pointed to an empty file)
3.  You don't give the user context: that they are opening a CSV file called xxx.csv, which will make it harder for them to know where the error comes from
4.  I can't count